src/test/java/test/parameters/issue1994/TestclassSample.java

```diff
@@ -0,0 +1,32 @@
+package test.parameters.issue1994;
+
+import org.testng.ISuite;
+import org.testng.ISuiteListener;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
+import org.testng.xml.XmlTest;
+
+@Listeners(TestclassSample.class)
+public class TestclassSample implements ISuiteListener {
+
+  public static int count = 0;
+
+  @BeforeClass
+  public void beforeClass(XmlTest xmlTest) {
+  }
+
+  @Test
+  public void testMethod() {
+  }
+
+  @Override
+  public void onFinish(ISuite suite) {
+    setCount(suite.getXmlSuite().getTests().size());
+  }
+
+  private static void setCount(int count) {
```

Comment:
What's the advantage instead of directly set `count`?

Reply:
I faintly remember a sonar warning which would advise against an instance method setting a static data member directly. That was why I chose to use a static method. There's no advantage whatsoever apart from readability I believe.

Reply:
ok